### ISSUE TITLE
Shared Deck calendar not found for PROPPATCH if shared without edit rights

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/deck#3422]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**

I have a bunch of Deck boards shared with me. Some are read-write, some are read-only. When I check the option to show Deck Calendars in the bottom left the boards show as calendars in Calendar. However, I can only toggle visibility for board calendars where I have edit rights.

**To Reproduce**
1. Share a calendar from user A to user B read-only
2. Open the Calendar app as user B
3. Toggle the calendar visibility
4. See an error

![Bildschirmfoto von 2021-11-11 17-50-20](https://user-images.githubusercontent.com/1374172/141336857-1e7e5653-8599-405d-a131-faee12857b0e.png)

**Expected behavior**

Visibility changes

**Screenshots**

see above

<details>
<summary>Server details</summary>

**Operating system**:

Arch, btw

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** 22.1 (company cloud)

---

This bug was originally identified by @AndyScherzinger. He's not able to toggle off the same board that I'm unable to toggle on.

**Comment by ChristophWurst** [comment]
I tried to understand why this could happen from reading the code of `\OCA\Deck\DAV\CalendarPlugin` but I don't see any difference between the two paths. If a calendar is listed for the calendar home then it should also be found when directly accessed.

**Comment by stefan-niedermann** [comment]
Isn't this a duplicate of https://github.com/nextcloud/deck/issues/2399 🤔?

**Comment by ChristophWurst** [comment]
No. I don't want to write to the calendar. I want to change one of its props. Those props go to caldav's custom property backend. This works find for most board, only the ones shared read-only don't work because the calendar node is not found.
